packages/react-avatar/SPEC-AvatarGroup.md

```diff
@@ -0,0 +1,155 @@
+# AvatarGroup Spec
+
+## Background
+
+The AvatarGroup component represents multiple people or entities. Avatar groups can be represented in a stacked layout, pie layout, and grid layout.
+
+## Prior Art
+
+### OpenUI
+
+There's no current research in OpenUI's website.
+
+| Library                                                                               | Name         | Notes                                                                                         |
+| ------------------------------------------------------------------------------------- | ------------ | --------------------------------------------------------------------------------------------- |
+| [Fluent UI v8](https://developer.microsoft.com/en-us/fluentui#/controls/web/facepile) | Facepile     | Receives list of `IFacepilePersona` to represent each avatar.                                 |
+| [Attlassian](https://atlassian.design/components/avatar-group/examples)               | Avatar group | Component uses data prop, which is a list of "entries" to represent each avatar in the group. |
+| [Ant Design](https://ant.design/components/avatar/)                                   | Avatar.Group | Uses subcomponent `<Avatar.Group>` to group the avatars given as children.                    |
+| [Primer React](https://primer.style/react/AvatarStack)                                | AvatarStack  | Acts similar to a FlexBox, avatars are given as children and grouped.                         |
+
+### Comparison v8 and v0
+
+There's only one existent component similar to AvatarGroup in v8 `Facepile`. v0 doesn't have an equivalent of this component.
+
+- v8 [Facepile](https://developer.microsoft.com/en-us/fluentui#/controls/web/facepile): Only offers grid layout and offers three overflow indicator styles.
+
+## Epic issue: [#22240](https://github.com/microsoft/fluentui/issues/22240)
+
+## Sample Code
+
+```jsx
+import { AvatarGroup, Avatar } from '@fluentui/react-avatar';
+
+const App = () => {
+  return (
+    <AvatarGroup layout="grid" size={32}>
+      <Avatar color="colorful" name="Katri Athokas" />
```

Comment:
AvatarGroup overrides its child `size` prope, should we also do that for `colorful`? I know `size` needs to be provided on AvatarGroup to provide consistent layout so maybe `colorful` can be conditional? E.g.,:

```jsx
<AvatarGroup color="colorful">
    <Avatar name="Elvia Atkins"/> // uses "color" from AvatarGroup
    <Avatar color="cornflower" name="Cameron Evans" /> // overrides "color" from AvatarGroup
</AvatarGroup>
```

Reply:
Based on the design spec, I have to use a specific order of colors for the avatars. I'll talk to design and see if doing this isn't a problem, but I totally agree with you.